### PR TITLE
getBillingPageToken resource for UserService

### PR DESCRIPTION
**Pull request by lounjukk**
getBillingPageToken service will return Recurly hostedLoginToken for user when available.

TODO:

- [x] Add all expected exceptions
- [x] Write integration tests

### CODE REVIEW

**Comment by lounjukk** [comment]
Dependent on https://github.com/lensapp/lenscloud/issues/1438

File: src/UserService.ts [hunk 1]
```diff
@@ -132,6 +137,17 @@ class UserService extends Base {
       },
     );
   }
+
+  async getBillingPageToken({ username }: { username: string }): Promise<BillingPageToken> {
+    const { apiEndpointAddress, fetch } = this.lensPlatformClient;
+    const url = `${apiEndpointAddress}/users/${username}/billing-page-token`;
+    const json = await throwExpected(
+      async () => fetch.get(url),
+      { 404: () => new NotFoundException(`Billing page token for ${username} not found`) },
```
Comment (DmitriyNoa) [inline src/UserService.ts:146]: Is 404 the only type of exception we will get from this endpoint?

Comment (lounjukk) [reply src/UserService.ts:146]: I think not. Waiting for implementation from @nevalla to add all relevant exceptions. I suppose we need at least 403 here as well.

Comment (lounjukk) [reply src/UserService.ts:146]: Added a TODO list to PR description to reflect this.

**Comment by lounjukk** [comment]
Depends on
https://github.com/lensapp/lenscloud/pull/1459

File: src/UserService.ts (hunk 1 above)
Comment (panuhorsmalahti) [reply src/UserService.ts]: FYI 403 is automatically mapped to `ForbiddenException`, but it's fine to have it here if we want to customize the Error message.

**Comment by lounjukk** [comment]
Closing this to create a new PR after rebase.

**Comment by panuhorsmalahti** [comment]
You could also rebase and force push

**Comment by lounjukk** [comment]
> You could also rebase and force push

Ok, will do that next time if that practice is ok here. Force with lease, I suppose?

**Comment by panuhorsmalahti** [comment]
Yeah, force with lease is better 👍

**Pull request closed without merging**